demux/packet.h

```diff
@@ -83,4 +89,19 @@ int demux_packet_add_blockadditional(struct demux_packet *dp, uint64_t id,
 
 void demux_packet_unref_contents(struct demux_packet *dp);
 
+struct demux_packet_pool {
+    mp_mutex lock;
+    struct demux_packet *packets;
+    struct MPContext *parent;
+};
+
+void demux_packet_pool_init(struct demux_packet_pool *pool, struct MPContext *parent);
+void demux_packet_pool_uninit(void *p);
+void demux_packet_pool_clear(struct demux_packet_pool *pool);
+void demux_packet_pool_add(struct demux_packet_pool *pool,
+                           struct demux_packet *dp);
+void demux_packet_pool_join(struct demux_packet_pool *pool,
+                           struct demux_packet *head, struct demux_packet *tail);
+struct demux_packet *demux_packet_pool_pop(struct demux_packet_pool *pool);
```

Comment:
the terminology here is kinda inconsistent
you have 'pop' like a stack, 'add' like a set and then 'join' like ??? which is actually the same as 'add'

suggestion: `pop`, `add`, `add_all` 
or maybe `new`, `free`, `free_all` to keep the allocator terminology?

Reply:
>  you have 'pop' like a stack, 'add' like a set and then 'join' like ??? which is actually the same as 'add'

It is mostly list. `join` was for joining two lists together. Chanted to perpend if that's better.

Changed to 

`push` for shorthand `push_front`
`pop` for shorthand `pop_front`
`prepend` for pretending list at front.